### PR TITLE
Feat: add voyage project submission status to `/me` endpoint

### PR DESCRIPTION
**Pull request by timDeHof**
# Description

## Features and Fixes
feat(users): Added support for the `projectSubmitted` field in the VoyageTeam entity and response.
fix(users): Updated the user object with sprintCheckInIds and projectStatus.
### Database Changes:
Added a new column `projectSubmitted` to the `VoyageTeam` table with a default value of false.
Updated `schema.prisma` to include the new `projectSubmitted` field in the `VoyageTeam` model.
### Backend Modifications:
Updated `users.select.ts` to include the `projectSubmitted` field in the select query for private user details.
Updated `team.entity.ts` to include the `projectSubmitted` field in the `VoyageTeamEntity` class.
Modified `users.response.ts` to include the `projectSubmitted` field in the `VoyageTeam` class.
Updated `users.service.ts` to fetch the `projectSubmitted` field from the database and update the user object with the sprintCheckInIds and projectStatus.

## Issue link

closes [Add voyage project submission status in `/me` endpoint](https://app.clickup.com/t/86b0dk58c)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Feature updates / changes
- [ ] Tests
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Testing can be done by inspecting the response for the `/me` endpoint in swagger for the field.It would be either "True" or "False".

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have updated the change log

### CODE REVIEW

**Review by cherylli — COMMENTED** [review]
we won't need projectSubmitted in the database as we can tell if the team has submitted the project if they have a record in the `formResponseVoyageProject` table

another thing is voyageTeam and teamMemberId are in `user.req` so we can save 2 two extra db calls

**Comment by timDeHof** [comment]
> we won't need projectSubmitted in the database as we can tell if the team has submitted the project if they have a record in the `formResponseVoyageProject` table
> 
> another thing is voyageTeam and teamMemberId are in `user.req` so we can save 2 two extra db calls

Ok, I was able to check and modify the voyageTeam with the `projectSubmitted` field to be set to true if the `formResponseVoyageProject` is not null in the voyageTeam. otherwise, it is set to false. This is now done without adding `projectSubmitted` in the database.

**Review by cherylli — APPROVED** [review]
thanks, this is perfect

File: src/users/users.service.ts [hunk 1]
```diff
@@ -76,6 +76,8 @@ export class UsersService {
             },
             select: privateUserDetailSelect,
         });
+        console.log("user", user);
```
Comment (cherylli) [inline src/users/users.service.ts:79]:
```suggestion

```

**Review by Ajen07 — APPROVED** [review]
Everything looks great !  The Swagger responses are functioning as expected.

**Review by cherylli — APPROVED** [review]

**Pull request merged**